include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -349,14 +349,25 @@ __max_compute_units(_ExecutionPolicy&& __policy)
 // Kernel run-time information helpers
 //-----------------------------------------------------------------------------
 
+// 20201214 value corresponds to oneAPI C++ Compiler Classic 2021.1.2 Patch release
+#if __SYCL_COMPILER_VERSION > 20201214
+#    define _USE_KERNEL_DEVICE_SPECIFIC_API 1
+#else
+#    define _USE_KERNEL_DEVICE_SPECIFIC_API 0
+#endif
```

Comment:
I think `#define _USE_KERNEL_DEVICE_SPECIFIC_API (__SYCL_COMPILER_VERSION > 20201214)` should also work.